src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -0,0 +1,409 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.resources;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLConnection;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.rest.AbstractResource;
+import org.geoserver.rest.AtomLink;
+import org.geoserver.rest.RestletException;
+import org.geoserver.rest.format.DataFormat;
+import org.geoserver.rest.format.ReflectiveHTMLFormat;
+import org.geoserver.rest.format.ReflectiveJSONFormat;
+import org.geoserver.rest.format.ReflectiveXMLFormat;
+import org.geoserver.rest.util.IOUtils;
+import org.geoserver.rest.util.RESTUtils;
+import org.restlet.Context;
+import org.restlet.data.MediaType;
+import org.restlet.data.Method;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+import org.restlet.data.Status;
+import org.restlet.resource.InputRepresentation;
+import org.restlet.resource.Representation;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+
+import freemarker.ext.beans.CollectionModel;
+import freemarker.template.Configuration;
+import freemarker.template.SimpleHash;
+
+/**
+ *
```

Comment:
@NielsCharlier as I indicated above, I understand why this class has that name, and recognise that you are being consistent. The name is nice and simple. I ask only for a few words in the class javadoc. Perhaps even using of ```{@link``` to make your job easier? Just something to indicate what it is to future maintainers. I am guessing you might add something like:
```
{@link org.restlet.resource.Resource} to provide operations on {@link org.geoserver.platform.resource.Resource} instances.
```